ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/castor/algorithm/AutomatonSimulator.java

```diff
@@ -368,11 +368,11 @@ Set<FlowCnec> gatherFlowCnecsForAutoRangeAction(RangeAction<?> availableRa,
                                                     Network network) {
         // UsageMethod should be FORCED
         if (availableRa.getUsageMethod(automatonState).equals(UsageMethod.FORCED)) {
-            if (availableRa.getUsageRules().stream().filter(usageRule -> usageRule instanceof OnInstant || usageRule instanceof OnContingencyState)
-                .anyMatch(usageRule -> usageRule.getUsageMethod(automatonState).equals(UsageMethod.FORCED))) {
+            if (availableRa.getTriggerConditions().stream().filter(triggerCondition -> triggerCondition.getCnec().isEmpty() && triggerCondition.getCountry().isEmpty())
```

Comment:
I'm not a fan of checking emptiness of attributes to know if an RA is applicable... we should discuss how to improve this design

Reply:
I agree, I just did this as a temporary solution for the behavior to be the same before and after the new code
But this is not elegant at all and I think there are some design discussions needed here

Reply:
I changed this by checking the type of TriggerCondition. If you think this is a better way to do, we could generalize the usage elsewhere in the code (mainly in tests).